src/Evolution/DiscontinuousGalerkin/LiftFromBoundary.hpp

```diff
@@ -0,0 +1,166 @@
+ * \f}
+ *
+ * where \f$\breve{\imath}\f$, \f$\breve{\jmath}\f$, and \f$\breve{k}\f$ are
+ * indices in the logical \f$\xi\f$, \f$\eta\f$, and \f$\zeta\f$ directions.
+ */
+template <size_t Dim, typename DtTagsList, typename BoundaryCorrectionTagsList>
+void lift_boundary_terms_gauss_points(
+    const gsl::not_null<Variables<DtTagsList>*> dt_vars,
+    const Scalar<DataVector>& volume_det_inv_jacobian,
+    const Mesh<Dim>& volume_mesh, const Direction<Dim>& direction,
+    const Variables<BoundaryCorrectionTagsList>& boundary_corrections,
+    const Scalar<DataVector>& magnitude_of_face_normal,
+    const Scalar<DataVector>& face_det_jacobian) noexcept {
+  const Mesh<Dim - 1> boundary_mesh =
+      volume_mesh.slice_away(direction.dimension());
+  const Mesh<1> slice_mesh = volume_mesh.slice_through(direction.dimension());
+  const size_t num_volume_grid_points = volume_mesh.number_of_grid_points();
+  const size_t num_boundary_grid_points = boundary_mesh.number_of_grid_points();
+  detail::lift_boundary_terms_gauss_points_impl(
+      make_not_null(dt_vars->data()), dt_vars->number_of_independent_components,
+      volume_mesh, direction.dimension(),
+      gsl::make_span(get(volume_det_inv_jacobian).data(),
+                     num_volume_grid_points),
+      num_boundary_grid_points,
+      gsl::make_span(boundary_corrections.data(), boundary_corrections.size()),
+      direction.side() == Side::Upper
+          ? Spectral::boundary_lifting_term(slice_mesh).second
+          : Spectral::boundary_lifting_term(slice_mesh).first,
+      magnitude_of_face_normal, face_det_jacobian);
+}
+
+/*!
+ * \brief Lift both the upper and lower (in logical coordinates) the boundary
+ * corrections to the volume time derivatives in the specified logical
+ * direction.
+ *
+ * The upper and lower boundary corrections in the logical direction given by
+ * `dimension` are lifted together in order to reduce the amount of striding
+ * through data that is needed and to improve cache-friendliness.
+ *
+ * The general lifting term is:
+ *
+ * \f{align*}{
+ * \partial_t u_{\alpha\breve{\imath}\breve{\jmath}\breve{k}}=\cdots
+ * -\frac{\ell_{\breve{\imath}}\left(\xi=1\right)}
+ *   {w_{\breve{\imath}}J_{\breve{\imath}\breve{\jmath}\breve{k}}}
+ *   \left[J\sqrt{
+ *   \frac{\partial\xi}{\partial x^i} \gamma^{ij}
+ *   \frac{\partial\xi}{\partial x^j}}
+ *   \left(G_{\alpha} + D_{\alpha}\right)
+ *   \right]_{\breve{\jmath}\breve{k}}\left(\xi=1\right)
+ * - \frac{\ell_{\breve{\imath}}\left(\xi=-1\right)}
+ *   {w_{\breve{\imath}}J_{\breve{\imath}\breve{\jmath}\breve{k}}}
+ *   \left[J\sqrt{
+ *   \frac{\partial\xi}{\partial x^i} \gamma^{ij}
+ *   \frac{\partial\xi}{\partial x^j}}
+ *   \left(G_{\alpha} + D_{\alpha}\right)
+ *   \right]_{\breve{\jmath}\breve{k}}\left(\xi=-1\right),
+ * \f}
+ *
+ * where \f$\breve{\imath}\f$, \f$\breve{\jmath}\f$, and \f$\breve{k}\f$ are
+ * indices in the logical \f$\xi\f$, \f$\eta\f$, and \f$\zeta\f$ directions.
```

Comment:
directions -> dimension

Reply:
I would also add that `G+D` correspond to the input parameter boundary_corrections and add a reference to the `Spectral::boundary_lifting_term` explaining that this computes that term from the volume_mesh and dimension.  Basically a way for the user to know how the input arguments of the function relate to the provided formulae.  Same comment for the function above